### PR TITLE
refactor event RPC to make a pure event

### PR DESCRIPTION
**Pull request by baichuan3**
resolve https://github.com/rooch-network/rooch/issues/969

Summary
1. refactor event RPC,  and remove `sender `, `tx_hash` and `timestamp`, to make a pure event
2. rename `parsed_event_data` to `encoded_event_data`, and unify with other Annotated state

Todo
1. How to raise an error inner Iterator::filter_map to avoid panic?

### CODE REVIEW

**Comment by baichuan3** [comment]
```
curl --location 'http://localhost:50051' \
--header 'Content-Type: application/json' \
--data '{
    "id": 101,
    "jsonrpc": "2.0",
    "method": "rooch_getEventsByEventHandle",
    "params": [
        "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent", null, "5"
    ]
}'
```


```
{
    "jsonrpc": "2.0",
    "result": {
        "data": [
            {
                "event": {
                    "event_id": {
                        "event_handle_id": "0xa91bc7d411afd987f3949d27e2eb6596c0b37859b115487713ec7e92eb07c6a1",
                        "event_seq": 0
                    },
                    "type_tag": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent",
                    "event_data": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c566400000000000000",
                    "event_index": 0
                },
                "decoded_event_data": {
                    "state": {
                        "value": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c566400000000000000",
                        "value_type": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent"
                    },
                    "move_value": {
                        "abilities": 0,
                        "type": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent",
                        "value": {
                            "addr": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56",
                            "amount": "100"
                        }
                    }
                }
            },
            {
                "event": {
                    "event_id": {
                        "event_handle_id": "0xa91bc7d411afd987f3949d27e2eb6596c0b37859b115487713ec7e92eb07c6a1",
                        "event_seq": 1
                    },
                    "type_tag": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent",
                    "event_data": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c566400000000000000",
                    "event_index": 0
                },
                "decoded_event_data": {
                    "state": {
                        "value": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c566400000000000000",
                        "value_type": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent"
                    },
                    "move_value": {
                        "abilities": 0,
                        "type": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent",
                        "value": {
                            "addr": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56",
                            "amount": "100"
                        }
                    }
                }
            },
            {
                "event": {
                    "event_id": {
                        "event_handle_id": "0xa91bc7d411afd987f3949d27e2eb6596c0b37859b115487713ec7e92eb07c6a1",
                        "event_seq": 2
                    },
                    "type_tag": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent",
                    "event_data": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c566400000000000000",
                    "event_index": 0
                },
                "decoded_event_data": {
                    "state": {
                        "value": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c566400000000000000",
                        "value_type": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent"
                    },
                    "move_value": {
                        "abilities": 0,
                        "type": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56::event_test::WithdrawEvent",
                        "value": {
                            "addr": "0x4b1b5d2a62f3c9a19a5314167439945bf98fbc871e1402e36d5bad3382064c56",
                            "amount": "100"
                        }
                    }
                }
            }
        ],
        "next_cursor": 2,
        "has_next_page": false
    },
    "id": 101
}
```

File: moveos/moveos-store/src/event_store/mod.rs [hunk 1]
```diff
@@ -138,10 +152,14 @@ impl EventDBStore {
         let mut iter = self.event_store.iter()?;
```
Comment (jolestar) [inline moveos/moveos-store/src/event_store/mod.rs:152]: Get events by handle needs to scan the whole event_store?

File: moveos/moveos-types/src/event.rs [hunk 2]
```diff
@@ -233,19 +231,19 @@ impl EventHandle {
 
 // #[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, JsonSchema)]
 // #[derive(Clone, Debug)]
-#[serde_as]
+// #[serde_as]
 #[derive(Eq, PartialEq, Clone, Debug)]
 pub struct MoveOSEvent {
```
Comment (jolestar) [inline moveos/moveos-types/src/event.rs:236]: MoveOSEvent = Event, remove the MoveOSEvent?

File: moveos/moveos-types/src/event.rs [hunk 3]
```diff
@@ -254,51 +252,27 @@ pub struct MoveOSEvent {
...
-            timestamp_ms,
-            // block_height,
-        }
+    pub fn new(event: Event) -> Self {
+        MoveOSEvent { event }
     }
 }
 
 #[derive(Debug, Clone)]
 pub struct AnnotatedMoveOSEvent {
```
Comment (jolestar) [inline moveos/moveos-types/src/event.rs:261]: Rename to AnnotateEvent?

File: moveos/moveos-store/src/event_store/mod.rs [hunk 4]
```diff
@@ -160,28 +178,28 @@ impl EventDBStore {
...
-            EventFilter::Sender(_sender) => {
-                return Err(anyhow!(
-                    "This type does not currently support filter combinations."
-                ));
-            }
+            // EventFilter::Sender(_sender) => {
+            //     return Err(anyhow!(
+            //         "This type does not currently support filter combinations."
+            //     ));
+            // }
```
Comment (jolestar) [inline moveos/moveos-store/src/event_store/mod.rs:189]: Commented out code can be cleaned up

**Review by jolestar — APPROVED** [review]
I will merge this PR first, @baichuan3 you can address the comment in the new PR.

**Pull request merged**